### ISSUE TITLE
[CT-2404] [Feature] add case-insensitivity config to accepted_values native test

### ISSUE THREAD

**Issue by fivetran-jamie** [dbt-labs/dbt-core#7344]
### Is this your first time submitting a feature request?

- [X] I have read the [expectations for open source contributors](https://docs.getdbt.com/docs/contributing/oss-expectations)
- [X] I have searched the existing issues, and I could not find an existing issue for this feature
- [X] I am requesting a straightforward extension of existing dbt functionality, rather than a Big Idea better suited to a discussion

### Describe the feature

I am adding an `accepted_values` test to the Fivetran Netsuite dbt package, as we have expected values hard-coded in a mapping model. thus, i want to raise a warning if there's an unexpected value that won't get mapped properly 

When comparing strings, we basically always do a case-insensitive match (ie `lower` both strings we're comparing). 

so, i am wondering if we can add a `case_sensitive` argument to accepted_values [macro](https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/include/global_project/macros/generic_test_sql/accepted_values.sql) that,  if `false` (i imagine the default would still be `true`), it would lowercase everything in the [where clause](https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/include/global_project/macros/generic_test_sql/accepted_values.sql#L16-L24)

### Describe alternatives you've considered

1. make my own custom test to do this
2. lowercase the string fields i'm looking at in the staging layer and apply the accepted_values test after this transformation 
3. cross my fingers and hope everything has consistent casing

### Who will this benefit?

People who like the simplicity of using the builtin dbt tests

### Are you interested in contributing this feature?

Yes

**Comment by dbeatty10** [comment]
Thanks for making this suggestion and the insight behind it @fivetran-jamie ! The number of [Fivetran packages](https://hub.getdbt.com/fivetran/) y'all build and maintain is truly amazing 🤩

I think either of the first two alternatives would be the best way to go on this one:
1. make your own custom test to do this
1. lowercase the string fields you're looking at in the staging layer and apply the `accepted_values` test after this transformation

We've minimized both the number of [out-of-the-box generic tests](https://docs.getdbt.com/reference/resource-properties/tests#out-of-the-box-tests) as well as their optional parameters. This keeps those four easy to use, document, and maintain.

Then for unique cases like yours, we've tried to make it easy for you to implement your desired logic in a custom generic test (or apply this testing further downstream).

Since there are a couple good options here, I'm going to close this as "not planned".
